tests/set_test.go

```diff
@@ -198,3 +215,56 @@ func TestWithKeepTTLFlag(t *testing.T) {
 
 	assert.Equal(t, out, fireCommand(conn, cmd), "Value mismatch for cmd %s\n.", cmd)
 }
+
+/*
+We open some connections to the db and fire concurrent SET commands for a particular key
+with different values. We expect that there are no dirty reads/writes.
+All the values read should be among the values that were attempted to set in the first place.
+*/
+func TestConcurrentSetCommands(t *testing.T) {
+	numOfConnections := 4
+	connectionValues := make(map[*net.Conn]*string)
+	expectedValues := make(map[string]struct{})
+	valuesReadChan := make(chan interface{}, numOfConnections)
+
+	// Create connections and the values to set through them.
+	for connNum := 0; connNum < numOfConnections; connNum++ {
+		value := testutils.GenerateRandomString(8, "abcdefghizklmopqrs12345")
```

Comment:
We don't need to introduce a random string generator, you can generate values using the index of the connection, or some combination of things. It allows us to make the test outputs more deterministic.

Reply:
Makes sense. Changing it.

Reply:
Done